### PR TITLE
Add SamrQueryInformationDomain2(Opnum 46)

### PR DESCRIPTION
**Pull request by sangdo-lee**
`SAMPRDomainLockoutInfo{lockoutDuration:-18000000000, lockoutObservationWindow:-18000000000,lockoutThreshold:0}`

image

### CODE REVIEW

File: src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrQueryInformationDomain2Request.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        return new SamrQueryInformationDomainResponse<T>(newDomainInformation(), infoLevel);
+    }
+
+    @Override
+    public void marshal(PacketOutput packetOut) throws IOException {
+        packetOut.write(handle.getBytes());
+        packetOut.writeShort(infoLevel.getInfoLevel());
+    }
+
+    public static class DomainLockoutInformation extends SamrQueryInformationDomain2Request<SAMPRDomainLockoutInfo> {
```
Comment (bli-r7) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrQueryInformationDomain2Request.java]: Is a static class like this necessary for this?

Comment (bli-r7) [reply src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrQueryInformationDomain2Request.java]: Feels a bit too verbose if we have to call it like below

`        SamrQueryInformationDomain2Request.DomainLockoutInformation request2 = new SamrQueryInformationDomain2Request.DomainLockoutInformation(`

Comment (sangdo-lee) [reply src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrQueryInformationDomain2Request.java]: @bli-r7 , for example, do you prefer the simplified code as following?
```
  public SamrQueryInformationDomainResponse<SAMPRDomainLockoutInfo> getDomainLockoutInfo(
        final DomainHandle DomainHandle) throws IOException {
       SamrQueryInformationDomain2Request request = new SamrQueryInformationDomain3Request(
            DomainHandle, DomainInformationClass.DOMAIN_LOCKOUT_INFORMATION);
        return transport.call(request);
    }
```
```
public class SamrQueryInformationDomain2Request
        extends RequestCall<SamrQueryInformationDomainResponse<SAMPRDomainLockoutInfo>> {
    public final static short OP_NUM = 46;
    private final DomainHandle handle;
    private final DomainInformationClass infoLevel;

    public SamrQueryInformationDomain3Request(final DomainHandle handle, final DomainInformationClass infoLevel) {
        super(OP_NUM);
        this.handle = handle;
        this.infoLevel = infoLevel;
    }

    @Override
    public void marshal(PacketOutput packetOut) throws IOException {
        packetOut.write(handle.getBytes());
        packetOut.writeShort(infoLevel.getInfoLevel());
    }

    @Override
    public SamrQueryInformationDomainResponse<SAMPRDomainLockoutInfo> getResponseObject() {
        return new SamrQueryInformationDomainResponse<SAMPRDomainLockoutInfo>(new SAMPRDomainLockoutInfo(), infoLevel);
    }

}
```

Comment (bli-r7) [reply src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrQueryInformationDomain2Request.java]: Yes.

File: src/test/java/com/rapid7/client/dcerpc/mssamr/messages/Test_SamrQueryInformationDomain.java [hunk 2]
```diff
@@ -57,6 +58,19 @@ public void SamrQueryLogOffInformationDomain() throws IOException {
...
+    @Test
+    public void SamrQueryLockoutInformationDomain() throws IOException {
+        String hexString = "000002000c00000000cc1dcffbffffff00cc1dcffbffffff0000";
+        SAMPRDomainLockoutInfo lockout = new SAMPRDomainLockoutInfo();
+        SamrQueryInformationDomainResponse<SAMPRDomainLockoutInfo> response = new SamrQueryInformationDomainResponse<>(
+            lockout, DomainInformationClass.DOMAIN_LOCKOUT_INFORMATION);
+        response.unmarshal(getPacketInput(hexString));
+
+        // 1800 seconds
+        assertEquals(-18000000000L, lockout.getLockoutDuration());
```
Comment (bli-r7) [inline src/test/java/com/rapid7/client/dcerpc/mssamr/messages/Test_SamrQueryInformationDomain.java]: Why is there a `-`

Comment (sangdo-lee) [reply src/test/java/com/rapid7/client/dcerpc/mssamr/messages/Test_SamrQueryInformationDomain.java]: @bli-r7 , FYI - https://github.com/rapid7/nexpose-r7shared/blob/9a73ebc4eb4752abb11592193fb065681886c080/src/main/java/com/rapid7/net/cifs/CifsUtil.java#L23-L59

**Pull request merged**